.github/workflows/pull-request-format.yml

```diff
@@ -0,0 +1,49 @@
+name: PR # Only a prefix to make it more readable in Github
+
+on:
+  push:
+    branches-ignore: [ master ]
+  pull_request:
+
+jobs:
+
+  clang-format:
+    name: Clang Format Check
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: 'recursive'
+
+      - name: Install clang format
+        run: |
+          sudo apt-get install -y clang-format-10
+
+      - name: Check the Formatting
+        run: |
+          ./formatcode.sh
+          ./CI/check-format.sh
+
+  checklist:
+    name: Checklist
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check Commit Message
+        uses: gsactions/commit-message-checker@v1
+        with:
+          pattern: '([a-zA-Z0-9\-_, ]*): (.{1,50})(\n){0,1}(\n.{1,72})*'
+          error: "The message of at least one of the commits not match the commit guidelines specified in CONTRIBUTING.rst"
```

Comment:
There are occasions where [the commit prefix contains a slash](https://github.com/obsproject/obs-studio/commit/ce6f233e2d026ea5373a74d245307492788346ad) (see also [one with a comma and slash](https://github.com/obsproject/obs-studio/commit/c021338ceb75e0e87b0cd6645ba8bf585686dc31)), or [the commit message subject contains the word "Revert" before the rest of the message follows in quotes](https://github.com/obsproject/obs-studio/pull/4641), or [the commit message has no prefix](https://github.com/obsproject/obs-studio/commit/90df8d44dfa71d8439c0b5e6d975dabfd17d93b2), or [the commit message subject has more than 50 characters after the prefix](https://github.com/obsproject/obs-studio/commit/ebbe856d83e5e96870a29c55686a5006d5c71113) because making it shorter makes it non-descriptive and our hard limit is 72 characters total.  All of these would fail to pass this check.

However, this check would, as far as I can tell, pass a commit message like this:
```
: No prefix, but colon still here

.
```

As this is, I don't think I can approve this PR unless all of the cases I mentioned above are handled gracefully.  Either the regex pattern needs to be adjusted, or a PR that fails this check needs to not hold up the rest of the workflow and simply be a warning.  Even in the case of the warning, I would prefer that the regex pattern handle the cases I've outlined above.